DevelopingSpace/starchart
# Improve race condition in npm run dev

This adds a 3s delay to help fix the race condition that happens when starting `dev` where the server isn't ready when Remix wants to start.

There is probably a better way to accomplish it, but I'm sick of having it not do the right thing every other run.

To test: `npm run dev` and you should have:

```
 npm run dev

> @senecacdot/starchart@0.6.0 predev
> npm run build:server


> @senecacdot/starchart@0.6.0 build:server
> esbuild --platform=node --format=cjs ./server.ts --outdir=build --bundle


  build/server.js  2.2mb ⚠️

⚡ Done in 45ms

> @senecacdot/starchart@0.6.0 dev
> cross-env NODE_ENV=development SECRETS_OVERRIDE=1 run-p dev:remix dev:server:delay


> @senecacdot/starchart@0.6.0 dev:remix
> remix watch


> @senecacdot/starchart@0.6.0 dev:server:delay
> run-s delay dev:server


> @senecacdot/starchart@0.6.0 delay
> node -e "setTimeout(() => process.exit(0), 3000)"

Watching Remix app in development mode...
💿 Built in 299ms

> @senecacdot/starchart@0.6.0 dev:server
> node --inspect --require ./node_modules/dotenv/config ./build/server.js

Debugger listening on ws://127.0.0.1:9229/04316e34-6cd4-4702-a967-5420c18857d7
For help, see: https://nodejs.org/en/docs/inspector
info: 🔌 setting up prisma client to 127.0.0.1:3306
debug: Expiration Requests init: adding jobs for certificate/record expiration
info: ✅ app ready: http://localhost:8080
info: process DNS record expiration
info: TODO: process certificate expiration
```
